agent/xpcbridge/xpc_bridge.c

```
@@ -108,12 +108,51 @@ int xpc_bridge_connect(const char *service_name, const void *context, xpc_bridge
+    // The earlier reservation marked in_use=1 outside this critical section
+    // for liveness; this second critical section commits the actual handles
+    // atomically with respect to send/disconnect.
+    pthread_mutex_lock(&g_slots_mutex);
     g_slots[handle].connection = conn;
     g_slots[handle].queue = queue;
+    pthread_mutex_unlock(&g_slots_mutex);
 
     return handle;
 }
 
+int xpc_bridge_send_application_control(int handle, const uint8_t *data, size_t len) {
```

Cppcheck (2.20.0)

[style] The function 'xpc_bridge_send_application_control' is never used.

(unusedFunction)